### ISSUE TITLE
Support the notion of "first" or "last" pages

### ISSUE THREAD

**Issue by pserwylo** [sverrirs/jekyll-paginate-v2#28]
It is quite common to have a paginator which, in addition to "next" and "previous" (already supported by `jekyll-paginate-v2`), also includes "first" and "last". I initially tried to hack this together using the `total_posts` variable, which _almost_ works, but not quite, for the following reasons:

* "First" button:
  * When `page == 1`, the button for page `1` is highilghted and unclickable.
  * On page 2 or higher, link the "first" button to `permalink | append: ../`
* "Last" button:
  * When `page == paginator.total_pages`, on the last page the button for page `total_pages` is highlighted and unclickable.
  * On every other page, link to `permalink: append: ../ | append: paginator.total_pages`.

The reason this _almost_ works is because when you are on all but the first page, then `permalink: append: ../` takes you to `/my-paginator-link/12/../50` (for current page 12, and total pages is 50).
However on the first page, it takes you to `/my-paginator-link/../50` which is incorrect, due to the lack of page number for page 1.

This can be worked around with lots of `if` statements, but it would be preferable if there was a `paginator.first_path` and `paginator.last_path` variable available.

**Comment by pserwylo** [comment]
FYI: [Here is the layout](https://gitlab.com/pserwylo/jekyll-fdroid/blob/640cc5a4349f4cb2f89ce94dcc3e37a468bd65a8/_includes/browse-paginator-navigation.html) we are using which works around the lack of "first" and "last".

Of interest are the first button:

```liquid
{% if include.paginator.page_trail[0].num > 1 %}
	<li class="nav page first">
		<a class="label" href="{{ include.permalink | append: '..' | prepend: site.baseurl }}">1</a>
	</li>
	<li class="nav disabled"><span class="label">&hellip;</span></li>
{% endif %}
```

and the last button:

```liquid
{% if include.paginator.page_trail[-1].num < include.paginator.total_pages %}
	<li class="nav disabled"><span class="label">&hellip;</span></li>
	<li class="nav page last">
		{% comment %} Workaround for https://github.com/sverrirs/jekyll-paginate-v2/issues/28 {% endcomment %}
		{% if include.paginator.page == 1 %}
			{% assign back = '' %}
		{% else %}
			{% assign back = '../' %}
		{% endif %}
		<a class="label" href="{{ include.permalink | append: back | append: include.paginator.total_pages | prepend: site.baseurl }}">{{include.paginator.total_pages}}</a>
	</li>
{% endif %}
```

**Comment by sverrirs** [comment]
This is a great idea, embarrassed that I didn't think of this in the first place :)

This issue has been fixed in v1.8.0 of the gem.
https://rubygems.org/gems/jekyll-paginate-v2/versions/1.8.0

Please refer to Example 03 for examples for the new pagination variables
https://github.com/sverrirs/jekyll-paginate-v2/blob/master/examples/03-tags/_layouts/home.html#L40

**Comment by pserwylo** [comment]
Thanks for the quick response and fix, much appreciated. Upon testing I unfortunately stumbled upon #28. However if I manually remove the offending code from that issue, then I can see that the newly added `first_page_path` and `last_page_path` works well. Cheers!
